### PR TITLE
Added rediraffe redirecgtion

### PR DESCRIPTION
**Pull request by NerdSec**
As discussed in #1865, added the ability to perform redirects outside of RTD by leveraging rediraffe.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -77,6 +78,63 @@ def setup(app):
...
+    "index-setup-jupyterhub.html": "jupyterhub/index.html",
+    "tools.html": "reference/tools.html",
+    "reference-docs.html": "reference/reference-docs.html",
+#    "index-reference.html": "reference.html",
+    "glossary.html": "reference/glossary.html",
+    "user-storage.html": "customizing/user-storage.html",
+    "user-resources.html": "customizing/user-resources.html",
+    "user-management.html": "customizing/user-management.html",
+    "user-environment.html": "customizing/user-environment.html",
+#    "index-customization-guide.html": "customization.html",
```
Comment (consideRatio) [inline doc/source/conf.py]: Why are these added as comments?

**Review by consideRatio — APPROVED** [review]
I tried this locally, and it works nice! I pushed a commit to make `make devenv` still work again after the refactoring.

I only wonder about the commented out redirects, why are they made? Ah they were an older redirect to a file that was moved in this recent refactoring. I updated them to reference directly to the new location.

**Comment by choldgraf** [comment]
wohoo! can we all also appreciate how great the rediraffe logo is?

**Comment by consideRatio** [comment]
Haha yes :D :1st_place_medal:
